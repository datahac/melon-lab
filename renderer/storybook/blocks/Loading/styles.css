.loading__item {
  display: inline-block;
  opacity: 0;
  margin-right: 1px;
}

.loading__invalid {
  color: var(--color-loss);
}

.loading__item:nth-child(1) {
  animation: opacitychange 1s ease-in-out infinite;
}

.loading__item:nth-child(2) {
  animation: opacitychange 1s ease-in-out 0.33s infinite;
}

.loading__item:nth-child(3) {
  animation: opacitychange 1s ease-in-out 0.66s infinite;
}

@keyframes opacitychange {
  0%,
  100% {
    opacity: 0;
  }
  50% {
    opacity: 1;
  }
}
