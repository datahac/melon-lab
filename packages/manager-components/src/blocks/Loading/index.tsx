import React, { Fragment, StatelessComponent } from 'react';

import styles from './styles.css';

export interface LoadingProps {
  dataAvailable?: boolean;
  loading?: boolean;
}

const Loading: StatelessComponent<LoadingProps> = ({
  children,
  dataAvailable = true,
  loading = true,
}) => {
  return (
    <span>
      <style jsx>{styles}</style>
      {loading ? (
        <Fragment>
          <span className="loading__item">.</span>
          <span className="loading__item">.</span>
          <span className="loading__item">.</span>
        </Fragment>
      ) : (
        <Fragment>
          {dataAvailable ? (
            <Fragment>{children}</Fragment>
          ) : (
            <span className="loading__invalid">{children}</span>
          )}
        </Fragment>
      )}
    </span>
  );
};

export default Loading;
