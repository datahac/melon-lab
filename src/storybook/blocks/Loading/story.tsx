import { storiesOf } from '@storybook/react';
import React from 'react';
import Loading from './index';

storiesOf('Blocks|Loading', module).add('Default', () => {
  return <Loading />;
});
