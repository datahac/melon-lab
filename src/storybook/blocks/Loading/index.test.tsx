import React from 'react';
import Loading from './index';

describe('Loading', () => {
  const defaultElement = <Loading />;
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(defaultElement);
  });

  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly without loading', () => {
    wrapper.setProps({ loading: false });
    expect(wrapper).toMatchSnapshot();
  });

  it('should render correctly without dataAvailable', () => {
    wrapper.setProps({ loading: false, dataAvailable: false });
    expect(wrapper).toMatchSnapshot();
  });
});
