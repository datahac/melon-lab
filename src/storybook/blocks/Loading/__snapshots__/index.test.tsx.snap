// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render correctly 1`] = `
<span
  className="jsx-__scopedHash"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <span
    className="jsx-__scopedHash loading__item"
  >
    .
  </span>
  <span
    className="jsx-__scopedHash loading__item"
  >
    .
  </span>
  <span
    className="jsx-__scopedHash loading__item"
  >
    .
  </span>
</span>
`;

exports[`Loading should render correctly without dataAvailable 1`] = `
<span
  className="jsx-__scopedHash"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
  <span
    className="jsx-__scopedHash loading__invalid"
  />
</span>
`;

exports[`Loading should render correctly without loading 1`] = `
<span
  className="jsx-__scopedHash"
>
  <JSXStyle
    css="__scoped"
    styleId="__scopedHash"
  />
</span>
`;
